ontology/rml-sourcetarget.owl

```diff
@@ -0,0 +1,267 @@
+                                         :Target
+                                       )
+                         ] ;
+             rdfs:range comp:Compression ;
+             rdfs:comment <http://w3id.org/rml/source-target/> ;
+             rdfs:label "compression" .
+
+
+###  http://w3id.org/rml/encoding
+:encoding rdf:type owl:ObjectProperty ;
+          rdfs:domain [ rdf:type owl:Class ;
+                        owl:unionOf ( :Source
+                                      :Target
+                                    )
+                      ] ;
+          rdfs:range enc:Encoding ;
+          rdfs:comment <http://w3id.org/rml/source-target/> ;
+          rdfs:label "encoding" .
+
+
+###  http://w3id.org/rml/iterator
+:iterator rdf:type owl:ObjectProperty ;
+          rdfs:domain :LogicalSource ;
+          rdfs:range rdf:Literal ;
+          rdfs:comment <http://w3id.org/rml/source-target/> ;
+          rdfs:label "iterator" .
+
+
+###  http://w3id.org/rml/logicalSource
+:logicalSource rdf:type owl:ObjectProperty ;
+               rdfs:domain :TriplesMap ;
+               rdfs:range :LogicalSource ;
+               rdfs:comment "Represents the logical source to be mapped. This can be a pointer to any dataset."@en ;
+               rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+               rdfs:label "logical source" .
+
+
+###  http://w3id.org/rml/logicalTarget
+:logicalTarget rdf:type owl:ObjectProperty ;
+               rdfs:domain :TermMap ;
+               rdfs:range :LogicalTarget ;
+               rdfs:comment "Represents the logical target to where triples are exported to. This can be a pointer to any dataset." ;
+               rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+               rdfs:label "logical target" .
+
+
+###  http://w3id.org/rml/null
+:null rdf:type owl:ObjectProperty ;
+      rdfs:domain :Source ;
+      rdfs:range rdf:Literal ;
+      rdfs:comment <http://w3id.org/rml/source-target/> ;
+      rdfs:label "null" .
+
+
+###  http://w3id.org/rml/query
+:query rdf:type owl:ObjectProperty ;
+       rdfs:domain :Source ;
+       rdfs:range rdf:Literal ;
+       rdfs:comment <http://w3id.org/rml/source-target/> ;
+       rdfs:label "query" .
+
+
+###  http://w3id.org/rml/referenceFormulation
+:referenceFormulation rdf:type owl:ObjectProperty ;
+                      rdfs:domain :LogicalSource ;
+                      rdfs:range :ReferenceFormulation ;
+                      rdfs:comment <http://w3id.org/rml/source-target/> ;
+                      rdfs:label "reference formulation" .
+
+
+###  http://w3id.org/rml/serialization
+:serialization rdf:type owl:ObjectProperty ;
+               rdfs:domain :LogicalTarget ;
+               rdfs:range formats:Format ;
+               rdfs:comment <http://w3id.org/rml/source-target/> ;
+               rdfs:label "serialization" .
+
+
+###  http://w3id.org/rml/source
+:source rdf:type owl:ObjectProperty ;
+        rdfs:domain :LogicalSource ;
+        rdfs:range :Source ;
```

Comment:
Can be anything as well. Example:

```
rml:source [ a rml:Source, dcat:Distribution;
         dcat:accessURL <file:///path/to/data.xml>;
     ];
```

Reply:
That's true, but they are necessarily a "source" right? I'm fine with the current range, but it might be confusing. I'm also fine with dropping the range.

Reply:
Yes, but the DCAT vocabulary does not have a triple `rdf:type rmls:Source` so not sure what ontology things will do then?

Reply:
Agree with Pano, within the open world assumption there'd be no problem as long as the instance is at least type of `rml:Source`

Reply:
Perfect! Specification matches this approach, resolving comment.